proposals/v3/ids.md

```diff
@@ -0,0 +1,204 @@
+machines in various topologies, it is critical that Synse be able to reliably identify a device
+across different service instances. This allows the captured time-series data to always be
+correctly associated with its source. 
+
+In Synse v2, the device ID alone is not globally unique, but in combination with the corresponding
+rack ID and board ID, it is -- hence the `rack/board/device` routing hierarchy.
+
+The [tag-based routing system](tags.md) introduced in Synse v3 will continue to support addressing
+devices individually, so the requirement for having a deterministic globally unique device ID
+remains unchanged. This document describes how such an identifier will be generated.
+
+## High Level Work Items
+- Add SDK configuration modeling for plugin namespacing
+- Add SDK support for parsing and verifying the namespace config
+- Update the logic for generating device IDs to include the plugin namespace
+
+## Proposal
+### Overview
+Each plugin will have a deterministic unique namespace, within which all deterministic device IDs
+will be generated. The combination of the two ensures that all devices IDs within a single Synse
+deployment will be globally unique. See also: [RFC4122 section 4.3](https://tools.ietf.org/html/rfc4122#section-4.3).
+
+The process of generating a unique deterministic ID for a device is already done by the plugin,
+so this document will not go into detail on that process.
+
+The process of assigning each plugin a unique namespace is the primary focus of this document.
+
+### Plugin Namespaces
+The source of a plugin namespace will depend on the deployment strategy of that plugin. This does
+not mean that it depends on what is running/managing the plugin (e.g. docker, kubernetes, systemd,
+something else, ...). It means that it depends on:
+- Where the plugin will run (bound to a single machine, schedule-able across multiple machines)
+- How many instances of the plugin will be running.
+
+To break down the use cases a bit further:
+
+1. The plugin is bound to the machine *(examples: runs as k8s DaemonSet, runs directly on host)*
+   - In this case, the machine ID (or any other machine-specific property, like
+     static IP) may be used as the primary component of the plugin namespace.
+     - *Caveat*: With plugins being run in containers, this information will need to passed
+       from the host to the container (ENV, volume mounting, ...)
+2. The plugin is not bound to a machine (can be scheduled anywhere) and will only have a **single**
+instance running at any given time.
+   - This is likely the most frequent use case.
+   - In this case, plugin metadata (e.g. the plugin *tag*) may be used as the primary component of
+     the plugin namespace.
+3. The plugin is not bound to a machine (can be scheduled anywhere) and may have any number
+of instances running at a given time (1..N). *(e.g. multiple IPMI plugins for different customers)*
+   - **TODO**
```

Comment:
Is there any updates on this?

Reply:
not yet.. still working through some ideas here, but I don't think that this will be a blocker right now. for example, it could be as simple as just requiring different instances to be configured with different identifiers/names.